src/widget/systemtrayicon.cpp

```diff
@@ -116,6 +113,26 @@ QString SystemTrayIcon::extractIconToFile(QIcon icon, QString name)
     return iconPath;
 }
 
+#if defined(ENABLE_SYSTRAY_GTK_BACKEND) || defined(ENABLE_SYSTRAY_STATUSNOTIFIER_BACKEND)
+GdkPixbuf* SystemTrayIcon::convertQIconToPixbuf(const QIcon &icon)
+{
+    void (*callbackFreeImage)(guchar*, gpointer) =
+        [](guchar*, gpointer image_bytes)
+    {
+        free(reinterpret_cast<guchar*>(image_bytes));
+    };
+    QImage image = icon.pixmap(64, 64).toImage();
+    if (image.format() != QImage::Format_RGBA8888_Premultiplied)
+        image = image.convertToFormat(QImage::Format_RGBA8888_Premultiplied);
+    guchar* image_bytes = (guchar*)malloc(image.byteCount());
```

Comment:
OK.  Another solution could be keeping the QImage around.  It would keep a little more memory around, but would avoid the copy.